# 
# Some Samsung DTV are really picky about proper http responses
# HEAD ... HTTP/1.0 need \r\n\r\n as EOF marker 
#
# Andreas Fiedler (smartsmurf@shared-files.de)
--- a/upnp/src/genlib/net/http/httpreadwrite.c	2017-11-14 12:02:53.000000000 +0100
+++ b/upnp/src/genlib/net/http/httpreadwrite.c	2018-05-04 03:41:05.984601521 +0200
@@ -1455,7 +1455,7 @@
 	membuffer_init(&membuf);
 	membuf.size_inc = (size_t)70;
 	/* response start line */
-	ret = http_MakeMessage(&membuf, response_major, response_minor, "RSCB",
+	ret = http_MakeMessage(&membuf, response_major, response_minor, "RSCBcc",
 			       http_status_code, http_status_code);
 	if (ret == 0) {
 		timeout = HTTP_DEFAULT_TIMEOUT;
@@ -1596,9 +1596,12 @@
 			    (http_major_version == 1 && http_minor_version == 1)
 			    ) {
 				/* connection header */
-				if (membuffer_append_str(buf, "CONNECTION: close\r\n"))
+				if (membuffer_append_str(buf, "CONNECTION: close"))
 					goto error_handler;
 			}
+			/* append \r\n in all cases, e.g. HTTP/1.0 */
+			if (membuffer_append_str(buf, "\r\n"))
+				goto error_handler;
 		} else if (c == 'N') {
 			/* content-length header */
 			bignum = (off_t) va_arg(argp, off_t);
